src/robotide/ui/tagdialogs.py

```diff
@@ -100,6 +102,15 @@ def _make_bindings(self):
         self._tags_list.Bind(wx.EVT_LIST_ITEM_RIGHT_CLICK, self.OnRightClick)
         self._tags_list.Bind(wx.EVT_LIST_COL_CLICK, self.OnColClick)
 
+    def _tag_name_for_sort(self, tag_name):
+        # Make tag_name lowercase for the sorting algorithm only
+        tag_name = tag_name.lower()
+        # Compare numbers as numeric values
+        # m = re.search(r"([^0-9]*)([0-9]*)", tag_name)
+        m = re.search(r"(\w*)([0-9]*)", tag_name)
+        g = m.groups()
```

Comment:
This version is actually even more buggy than the old one. Now tags that contain special characters like `foo-1` are not handled correctly.